### PR TITLE
MTKA 1434 Email Publisher UI

### PR DESCRIPTION
**Pull request by andrewbotz**
## Description

Create the initial UI for the publisher side that includes Email input.

https://wpengine.atlassian.net/browse/MTKA-1434

## Testing

1. Create model with an Email type input from
2. Attempt to publish with step 1's model that includes email

## Screenshots

Screen Shot 2022-03-22 at 10 17 28 AM
Screen Shot 2022-03-22 at 10 17 43 AM
Screen Shot 2022-03-22 at 10 18 03 AM
Screen Shot 2022-03-22 at 10 18 29 AM

### CODE REVIEW

File: tests/acceptance/CreateContentModelEmailFieldCest.php [hunk 1]
```diff
@@ -0,0 +1,23 @@
+<?php
+
+class CreateContentModelEmailFieldCest {
+	public function i_can_create_a_content_mdodel_email_field( AcceptanceTester $i ) {
```
Comment (nickcernis) [inline tests/acceptance/CreateContentModelEmailFieldCest.php:4]: nit: typo

```suggestion
	public function i_can_create_a_content_model_email_field( AcceptanceTester $i ) {
```

Comment (nickcernis) [inline tests/acceptance/CreateContentModelEmailFieldCest.php:4]: nit: typo

```suggestion
	public function i_can_create_a_content_model_email_field( AcceptanceTester $i ) {
```

File: tests/acceptance/PublishModelCest.php [hunk 2]
```diff
@@ -169,6 +169,29 @@ public function i_see_submission_errors_in_number_fields_when_input_is_more_than
 		$i->seeInField( 'atlas-content-modeler[goose][decimal]', '20' );
 	}
 
+	public function i_see_submission_errors_in_eamil_field_when_input_is_not_a_email( \AcceptanceTester $i ) {
```
Comment (nickcernis) [inline tests/acceptance/PublishModelCest.php:172]: nit: typo

```suggestion
	public function i_see_submission_errors_in_email_field_when_input_is_not_an_email( \AcceptanceTester $i ) {
```
```suggestion
	public function i_see_submission_errors_in_eamil_field_when_input_is_not_a_email( \AcceptanceTester $i ) {
```

Comment (nickcernis) [inline tests/acceptance/PublishModelCest.php:172]: nit: typo

```suggestion
	public function i_see_submission_errors_in_email_field_when_input_is_not_an_email( \AcceptanceTester $i ) {
```
```suggestion
	public function i_see_submission_errors_in_eamil_field_when_input_is_not_a_email( \AcceptanceTester $i ) {
```

**Review by nickcernis — COMMENTED** [review]
Lookin' good!

Forgive me if this is part of upcoming work, but do you think we should improve the error messaging in the case of an invalid email?

In Chrome, if you return to the form to enter an invalid email address in a non-required email field after clicking publish you get, “This field is required”. That seems a little confusing when the field is not required, and when the field has characters in it (but no valid email address yet):

https://user-images.githubusercontent.com/647669/159566356-d9681e99-4f2c-4e30-b844-33703c5783b9.mov

Is it worth displaying “Email address invalid” or similar instead in that case?

This at the top of the `validate()` function in `includes/publisher/js/src/components/Field.jsx` seems to work, but may need more testing:

```js
		if (field.type === "email") {
			if (!event.target.validity.valid) {
				error = __("Email address invalid.", "atlas-content-modeler");
			}

			if (field?.required && event.target.value.trim() === "") {
				error = defaultError;
			}
		}
```


https://user-images.githubusercontent.com/647669/159567486-c66ea85c-28d0-46a6-a604-43c9b45131d7.mov

**Review by mindctrl — COMMENTED** [review]
Aside from the issues Nick mentioned, this is looking good!

**Review by mindctrl — APPROVED** [review]
This looks good! Validation stuff to go in another PR.

**Pull request merged**